lib/graphql/lang/ast/type_info_visitor.ex

```diff
@@ -127,7 +127,7 @@ defmodule GraphQL.Lang.AST.TypeInfoVisitor do
         :ObjectField ->
           input_type = TypeInfo.input_type(accumulator[:type_info])
           object_type = TypeInfo.named_type(accumulator[:type_info], input_type)
-          if object_type === %GraphQL.Type.ObjectType{} do
+          if %GraphQL.Type.ObjectType{} = object_type do
```

Comment:
@aweiker I agree this could be cleaner, but I want to keep the structure in line with the JS implementation until all the validations are ported over. Once we're validations-complete, `TypeInfo` can be refactored to make it more Elixir-y. In the meantime, it mirrors the JS implementation in structure which makes it easier to reason about when things aren't working. Fair enough?